src/client/components/WidgetContainer/widgets/EditoriallySelectedMaterialsWidget/EditoriallySelectedMaterials.widget.component.js

```diff
@@ -49,21 +51,31 @@ export class EditoriallySelectedMaterialsWidget extends AbstractWidget {
   }
 
   render() {
-    const works = (this.props.widgetReducerProp.works[this.state.identifier] || []).slice(0, this.state.closed ? 6 : (this.state.works.length));
+    const works = (this.props.widgetReducerProp.works[this.state.identifier] || []).slice(0, this.state.closed ?
+      6 :
+      (this.state.works.length));
 
     let closeButtonContent;
-    if (this.state.closed) {
+    if (works.length <= 5) {
+      closeButtonContent = null;
+    }
+    else if (this.state.closed) {
       closeButtonContent = (
-        <span>
-        <Icon glyph={plusSvg} /> VIS FLERE
-      </span>
+        <a className="editorially-selected-materials-widget--show-more-button" onClick={() => this.setState({closed: !this.state.closed})}>
+          <span>
+            <Icon glyph={plusSvg} /> VIS FLERE
+          </span>
+        </a>
+
       );
     }
     else {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either merge this branch with the identical one on line "62" or change one of the implementations. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1871)